### ISSUE TITLE
Improve error message when web console can't display logs

### ISSUE THREAD

**Issue by spadgett** [openshift/origin#5913]
This error message implies that something went wrong, although it usually just means the logs are no longer available. If we don't know the real cause of the error, we might use a more generic message like "Could not load logs." I also don't like the big, empty black box with no content.

<img width="438" alt="openshift_web_console" src="https://cloud.githubusercontent.com/assets/1167259/11182187/dbc1cec6-8c36-11e5-8413-9debf054e817.png">

/cc @jwforres @benjaminapetersen @ajacobs21e

**Comment by benjaminapetersen** [comment]
NOTE: check out [this issue](https://github.com/openshift/origin/pull/6878#issuecomment-176281046) for some comments on adding messages, such as when a build status is `Error` (comment was left in code).

**Comment by benjaminapetersen** [comment]
Couple options for how we could display error messages:

- A user is probably already looking for the log output, so printing the error here will ensure it is read:
![screen shot 2016-02-05 at 3 12 05 pm](https://cloud.githubusercontent.com/assets/280512/12857974/306a794a-cc1b-11e5-9878-ecc1cf3d9cf4.png)

- Or we can reuse an alert box style in place of the log viewer.  This may be initially disorienting to a user until it is experienced a few times, but is probably fine:
![screen shot 2016-02-05 at 3 13 32 pm](https://cloud.githubusercontent.com/assets/280512/12857977/35e27be8-cc1b-11e5-874a-68cd0c94f78b.png)

If we go with the second approach but already have an alert on the page (top) is it weird?
![screen shot 2016-02-05 at 3 18 47 pm](https://cloud.githubusercontent.com/assets/280512/12858104/03304850-cc1c-11e5-8a40-e8f7546dc42b.png)

Open to thoughts and suggestions from anyone.  @ajacobs21e @sg00dwin

**Comment by benjaminapetersen** [comment]
Per comment above, we prob should have an alert style message (or something like it) to handle 'logs not able to run yet':
```javascript
 $scope.logCanRun = !(_.includes(['New', 'Pending', 'Error'], build.status.phase));
```

**Comment by spadgett** [comment]
The first screenshot you have looks best, but I'm afraid it would be confused with an actual error in a log itself instead of an error loading the log.

I'd like a style for errors when an entire page or tab doesn't load that we can use everywhere. We don't  handle that very well today.

/cc @ajacobs21e

**Comment by jwforres** [comment]
agree with @spadgett the first one looks like an error coming back in the log itself

**Comment by benjaminapetersen** [comment]
Hmm. Yeah, not sure the best approach here.

**Comment by benjaminapetersen** [comment]
How about alert within the log?

![screen shot 2016-02-05 at 3 18 24 pm-boxes](https://cloud.githubusercontent.com/assets/280512/12963108/cd04c6a8-d016-11e5-930e-6f1cb3094a8f.png)
boxes. :smile:

**Comment by spadgett** [comment]
I was thinking of something along the lines of our empty state message.

<img width="786" alt="openshift_web_console" src="https://cloud.githubusercontent.com/assets/1167259/12964670/66f52c6e-d01f-11e5-9952-f92490c61e5f.png">

**Comment by benjaminapetersen** [comment]
+1 to that, much better!  Others?  Thoughts?

**Comment by spadgett** [comment]
@ajacobs21e thoughts on this?

**Comment by benjaminapetersen** [comment]
Agree, with @ajacobs21e's :+1: I'll make the change to this.

**Comment by spadgett** [comment]
By the way, it's just the `empty-state-message` style with centered text and adjusted top margin.

**Comment by spadgett** [comment]
Example here: https://github.com/openshift/origin/blob/master/assets/app/views/browse/build-config.html#L72-L89

**Comment by ajacobs21e** [comment]
@benjaminapetersen I'm here finally!

I agree with Sam's empty state style message however I suggesting using the (i) info icon. The warning looks like something is wrong, which I don't believe is the case if it's just that the log is archived.

Are we able to tell the difference between archived and something wrong? "could not be loaded" sounds like I did something wrong or something else went wrong that I need to go fix. 

The first screenshot had a reload option. What's the use case for that? versus "view archive"?

**Comment by ajacobs21e** [comment]
although I do think 

> no logs available. sad

is adorable :)

**Comment by benjaminapetersen** [comment]
Btw, if Kibana is not setup, there will not be the blue button.  At that point, this could be a bit bland:

<img width="700" alt="screen shot 2016-02-15 at 4 29 02 pm" src="https://cloud.githubusercontent.com/assets/280512/13060701/4425c2bc-d401-11e5-8d6a-30bf90b66231.png">

**Comment by spadgett** [comment]
> Are we able to tell the difference between archived and something wrong? "could not be loaded" sounds like I did something wrong or something else went wrong that I need to go fix.

No, that's the problem. We can't get the error details. The most common reason is that the pod has been deleted and logs are no longer available, but it could also be an error connecting to the server.

You should see this message less in the future. We'll be able to show logs for terminated containers and complete deployments.

**Comment by spadgett** [comment]
> The first screenshot had a reload option. What's the use case for that? versus "view archive"?

Reload will try to load the log again. It's probably less useful now that we don't attempt to request the logs when a pod is still pending.

View Archive is a link to Kibana where the logs are archived for pods that are no longer available.

**Comment by benjaminapetersen** [comment]
So working out the details here:
- [ ] eliminate 'Current log from....' if there is no log. 
- [ ] move status to above error message & fix weird spacing by "Cancelled" and icon (in image above)
- [ ] handle `.onError` w/an appropriate empty state message (connection fail, etc)
- [ ] **PRIMARY**: handle `.onClose` when no log data has been sent.  This is the main issue we can deal with here.  The pod is gone, nothing went wrong, but there is no additional info for hte user
- [ ] show link to Kibana if Kibana has been setup.

**Comment by ajacobs21e** [comment]
@benjaminapetersen what's the "status: canceled" text in your latest screenshot?

@spadgett do you think it's worth trying to rephrase "could not be loaded" ? It's still bothering me that it sounds like either I did something wrong or that there's a fix available. But right now I can't think of another phrasing. I could try if I'm not the only one bothered by it.

**Comment by benjaminapetersen** [comment]
@ajacobs21e you mean [this one](https://github.com/openshift/origin/issues/5913#issuecomment-184403590)?  That screen shot is out of date.  The latest are actually on [the PR that will fix this issue](https://github.com/openshift/origin/pull/7363#discussion_r53244717).

Let me know if that clarifies!

**Comment by jwforres** [comment]
Fixed by #7363
